### ISSUE TITLE
Refactor testkit into futures/promises

### ISSUE THREAD

**Issue by torkelrogstad** [bitcoin-s/bitcoin-s#307]
`Unit`-> `Future[Unit]`, `FooClient` -> `Future[FooClient]`

**Comment by Christewart** [comment]
Eclair stuff was taken care of in #328 , however the bitcoind testkit still needs work

**Comment by torkelrogstad** [comment]
This is done in the `rpc_version` feature branch, issue can be closed once that branch is merged into master.
